src/services/AdvancedTradingView/adapters/baseAdapter/base.adapter.ts

```diff
@@ -121,20 +122,99 @@ export class BaseAdapter<
     }
   }
 
+  public async getPairData({
+    dataType,
+    inputToken,
+    outputToken,
+    amountToFetch,
+    isFirstFetch,
+    abortController,
+    refreshing,
+  }: AdapterFetchDetailsExtended) {
+    if (!this.isSupportedChainId(this._chainId)) return
+
+    const pairId = this._getPairId(inputToken, outputToken)
+
+    if (!pairId) return
+
+    const pair = this.store.getState().advancedTradingView.adapters[this._key][pairId]
+
+    // TODO: CLARIFY THIS WITH ADAM
+    // if (
+    //   (dataType === AdapterPayloadType.SWAPS && pair && !isFirstFetch && !pair.swaps?.hasMore) ||
+    //   (dataType === AdapterPayloadType.SWAPS && pair && isFirstFetch) ||
+    //   (dataType === AdapterPayloadType.BURNS_AND_MINTS && pair && !isFirstFetch && !pair.burnsAndMints?.hasMore) ||
+    //   (dataType === AdapterPayloadType.BURNS_AND_MINTS && pair && isFirstFetch)
+    // )
+    //   return
```

Comment:
@bejzik8 context 😅?

Reply:
I am not sure if this is needed or not, it was there and I was thinking you might be able to give me some context and clarify if this is necessary or not. 😅